manim/mobject/vector_field.py

```diff
@@ -208,11 +208,54 @@ def construct(self):
         """
 
         step_size = dt / substeps
+
+        def runge_kutta_pointwise(self, p: Sequence[float], step_size: float) -> float:
```

Comment:
It seems a little strange to have the RK function defined inside the `nudge()` definition. That means it will be redefined every time nudge() is called, despite the definition never changing. 

But also, the RK function itself looks like a standalone function. It could take the `self.func` function as an argument, rather than `self`, to be more general. And if decoupled like that, perhaps it belongs in some utils/ file?